lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -429,10 +478,15 @@ performSelection minCoinFor costFor bundleSizeAssessor criteria
     | otherwise = do
         state <- runSelection
             selectionLimit extraCoinSource utxoAvailable balanceRequired
-        let balanceSelected = fullBalance (selected state) extraCoinSource
-        if balanceRequired `leq` balanceSelected then
+        let
+          balanceSelected =
+              fullBalance (selected state) extraCoinSource
+                  `TokenBundle.add`
+                      TokenBundle.fromTokenMap mintInputs
```

Comment:
I suspect you don't need to add the `mintInputs` here, as we just need to check that what we've selected is greater than what was required, which already has the minted value removed.